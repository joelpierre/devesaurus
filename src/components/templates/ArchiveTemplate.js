import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import Heading from '../core/Heading/Heading';
import { CoreLayout } from '../../layouts/CoreLayout';

export class ArchiveTemplate extends PureComponent {
  render() {
    const { pageContext } = this.props;

    return (
      <CoreLayout
        title={pageContext.name}
        description=""
        data-test="template-archive"
      >
        <Heading priority="1" classes="text-center">
          {pageContext.name}
        </Heading>
      </CoreLayout>
    );
  }
}

ArchiveTemplate.defaultProps = {
  pageContext: null,
};

ArchiveTemplate.propTypes = {
  pageContext: PropTypes.instanceOf(Object),
};

export default ArchiveTemplate;
