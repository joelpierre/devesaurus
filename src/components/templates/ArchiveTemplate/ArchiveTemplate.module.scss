@import "~sass/core";

.archive {
  padding: 0;
  margin: 0;

  &__word-category {
    visibility: visible;
  }

  &__word-tag {
    visibility: visible;
  }

  &__category {
    visibility: visible;
  }

  &__post-tag {
    visibility: visible;
  }
}
