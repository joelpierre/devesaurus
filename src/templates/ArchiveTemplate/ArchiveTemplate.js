import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import Heading from '../../components/core/Heading/Heading';
import { CoreLayout } from '../../layouts/CoreLayout';
import Section from '../../components/core/Section/Section';
import Container from '../../components/core/Container/Container';
import Row from '../../components/core/Row/Row';
import Flex from '../../components/core/Flex/Flex';
import styles from './ArchiveTemplate.module.scss';

export class ArchiveTemplate extends PureComponent {
  render() {
    const { pageContext } = this.props;

    console.log(pageContext);

    return (
      <CoreLayout
        title={pageContext.name}
        description=""
        data-test="template-archive"
        classes={classNames([
          styles.archive,
          styles[`archive__${pageContext.taxonomy.replace('_', '-')}`],
        ])}
      >
        <Section>
          <Container>
            <Row>
              <Flex>
                <Heading priority="1" classes="text-center mb-0">
                  {pageContext.name}
                </Heading>
              </Flex>
            </Row>
          </Container>
        </Section>
      </CoreLayout>
    );
  }
}

ArchiveTemplate.defaultProps = {
  pageContext: null,
};

ArchiveTemplate.propTypes = {
  pageContext: PropTypes.instanceOf(Object),
};

export default ArchiveTemplate;
