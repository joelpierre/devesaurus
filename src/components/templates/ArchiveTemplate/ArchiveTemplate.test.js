import React from 'react';
import { shallow } from 'enzyme/build';
import toJson from 'enzyme-to-json';

import ArchiveTemplate from './ArchiveTemplate';
import {
  checkProps,
  findByTestAttr,
  matchSnapshot,
} from '../../../utils/test-utilities';

const defaultProps = {
  pageContext: {
    name: 'test',
    slug: 'test-slug',
    taxonomy: 'test-taxonomy',
  },
};

/**
 * Factory function to create a ShallowWrapper for the ArchiveTemplate component.
 * @param {object} props - Component props specific to setup
 * @returns {ShallowWrapper}
 */
const setup = (props = {}) => {
  const setupProps = { ...defaultProps, ...props };
  return shallow(<ArchiveTemplate {...setupProps} />);
};

describe('<ArchiveTemplate/>', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = setup();
  });

  it('renders the ArchiveTemplate Component without errors', () => {
    const template = findByTestAttr(wrapper, 'template-archive');
    expect(template).toBeTruthy();
    expect(template.length).toBe(1);
  });

  it('should render with correct props', () => {
    checkProps(ArchiveTemplate, defaultProps);
  });

  it('should match snapshot', () => {
    matchSnapshot(wrapper);
  });
});
