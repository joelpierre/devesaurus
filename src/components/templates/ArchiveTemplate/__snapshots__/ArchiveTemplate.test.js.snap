// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArchiveTemplate/> should match snapshot 1`] = `
<CoreLayout
  classes="archive archive__test-taxonomy"
  data-test="template-archive"
  description=""
  onGetSiteMeta={[Function]}
  onGetSiteOptions={[Function]}
  title="test"
>
  <Section
    classes={null}
    contrast={false}
    theme={null}
    type="section"
  >
    <Container
      fluid={false}
    >
      <Row
        column={false}
        reverse={false}
        row={true}
      >
        <Flex
          classes={null}
          col={null}
          colLg={null}
          colMd={null}
          colSm={null}
          colXl={null}
          colXs={null}
          lg={false}
          md={false}
          sm={false}
          type="div"
          xl={false}
          xs={false}
        >
          <Heading
            classes="text-center mb-0"
            priority="1"
          >
            test
          </Heading>
        </Flex>
      </Row>
    </Container>
  </Section>
</CoreLayout>
`;
